### ISSUE TITLE
Support for `cloudflare_worker_script` resources being module syntax

### ISSUE THREAD

**Issue by kayteh** [cloudflare/terraform-provider-cloudflare#1417]
### Current Terraform and Cloudflare provider version

```
$ terraform -v
Terraform v1.1.4
on linux_amd64
+ provider registry.terraform.io/cloudflare/cloudflare v3.8.0
```

### Description

Similar to the proposed upstream change (cloudflare/cloudflare-go#794), `cloudflare_worker_script` resource needs to be aware of Worker module syntax. The quick tl;dr: Workers only accepts module-based workers when uploaded with `Content-Type: application/javascript+module` header.

It seems most worth it to add a `module` boolean parameter, and pass through through to cloudflare-go.

Attempting to upload a valid worker without this sort of change results in a response: 
```json
{
  "code": 10021,
  "message": "Uncaught SyntaxError: Unexpected token 'export'\n  at line 1\n"
}
```

### Use cases

Uploading workers scripts based on ES Modules, documented here: https://developers.cloudflare.com/workers/learning/migrating-to-module-workers

### Potential Terraform configuration

```hcl
resource "cloudflare_worker_script" "hello" {
  name    = "hello"
  content = "export default { async fetch() { return new Response('hello-world') } }"
  module  = true
}
```

### References

- cloudflare/cloudflare-go#794

**Comment by sodabrew** [comment]
My $0.02 suggestion:

Instead of:
```
resource "cloudflare_worker_script" "hello" {
  name    = "hello"
  content = "export default { async fetch() { return new Response('hello-world') } }"
  module  = true   # If false, then type worker is implied
}
```

How about:
```
resource "cloudflare_worker_script" "hello" {
  name    = "hello"
  content = "export default { async fetch() { return new Response('hello-world') } }"
  type  = "module" # default is "worker"
}
```

**Comment by moritonal** [comment]
So.. can anyone confirm the "correct" way to deploy CloudFlare Workers with Typescript via Terraform?

Currently I have a fairy hacky start of a solution where I have to use `npx wrangler publish --outdir dist --dry-run` to build the compiled script, then use `content = file("dist/index.js")` in Terraform, only to get `Uncaught SyntaxError: Unexpected token 'export'` which actually means that the provider didn't add a header during upload?

**Comment by dotansimha** [comment]
Same here. Seems like the actual change the needs to be done is in the `Content-Type`.

### Sending a regular worker code (event handler) 

![image](https://user-images.githubusercontent.com/3680083/182547284-9e93f7f1-cb06-439d-976e-7fa84adf9d31.png)

`Content-Type` is set to `application/javascript`

### Sending a `module`-based worker

![image](https://user-images.githubusercontent.com/3680083/182547312-e92ebc89-87ea-46d0-ac48-b448986ced93.png)

`Content-Type` is set to `application/javascript+module`

**Comment by dotansimha** [comment]
It seems like https://github.com/cloudflare/cloudflare-go/issues/794 fixed this issue, and we might be able to use it here soon?

**Comment by sodabrew** [comment]
@dotansimha it's halfway there -- uploading a module script works but downloading it returns a raw multipart/form-data body, which makes it impossible to compare the terraform state and determine whether the resource has changed. See https://github.com/cloudflare/cloudflare-go/pull/1040 for my personal first effort at a solution, but there isn't an official direction on this yet, nor on how to handle a multiscript bundle.

**Comment by sodabrew** [comment]
https://github.com/cloudflare/cloudflare-go/pull/1040 is landed! With cloudflare-go 0.49.0, it will be possible to both push a module-format worker script and also download one, enabling the terraform state refresh and comparison steps.

Multiple-script workers are still an open issue that isn't fully explored, but I probably won't be pushing through solving that.
